### DISCUSSION TITLE
Stop hiding conversations in PRs

### DISCUSSION THREAD

**Discussion by jorisw** [community/community#130618] [Pull Requests]
### Select Topic Area

Product Feedback

### Body

![image](https://github.com/community/community/assets/30017/51b6e79d-dd1f-425d-9d83-52e69ca99ecf)

This very strange pattern at GitHub pull requests is disrupting smooth reviews between members of my team. This marker is easily missed and leading developers to re-submit review requests before having addressed all concerns.

Please remove this 'feature' so that we're sure that our feedback is seen by our peers.

Note: This is _not_ about the amount of messages in a thread. This is about individual comments on code being hidden once they're over 10 or so.

**Comment by chavlji** [comment]
💯% agree!

**Reply by danielsn** [reply]
200% agree

**Reply by danielsn** [reply]
At the very least, if conversations must be hidden, hide the resolved one so we can see what unresolved issues remain.

**Comment by bugfood** [comment]
This appears to duplicate https://github.com/orgs/community/discussions/112603.

**Reply by medmunds** [reply]
It is also a (near-)duplicate of https://github.com/orgs/community/discussions/11842 from 2022 (albeit with a more actionable title). And https://github.com/orgs/community/discussions/75820 proposes GitHub work around the problem by showing the number of unresolved items in the "hidden" message. Upvotes all around.

(I've started adding instructions to "please search this page for 'hidden conversations'" in every code review. 😬 )

**Reply by jorisw** [reply]
> (I've started adding instructions to "please search this page for 'hidden conversations'" in every code review. 😬 )

Same here.

**Reply by ChrisMacGregorONI** [reply]
And there's https://github.com/orgs/community/discussions/7638 as well...  seems to be a popular/frequent complaint.

**Comment by ChrisMacGregorONI** [comment]
Alternately, put a button on the page to click to "Show all conversations" (or "Show all unresolved conversations") which unhides ALL of them so I don't have to click several times.  (Note that it can be hard to keep track of where you are in a long sequence of them (esp. if some are resolved) because of the visual "jump" when you un-hide some.)  So if you (GH devs/PMs) are convinced that the current behavior is preferred by the majority (or should be the default for some other reason), then please at least give the rest of us a one-click way to make sure we're not missing anything.

Or at least add a Settings option somewhere to disable this, or to set the threshold for it.

Authors of PRs I've reviewed have inadvertently merged code with active unresolved comments simply because they missed comments due to this.  (And yes, it really was inadvertent.)

Hiding unresolved comments seems to defeat the very purpose of review comments; I would think that unresolved comments would be considered one of the highest-priority things on the page!  Can you please share the rationale for hiding them by default?

**Reply by bugfood** [reply]
Very much agreed. I need to remind PR authors to look for the hidden conversations. If a PR author hasn't addressed a comment, then I wonder if they even saw the comment. Hiding conversations is a misfeature and we should not need to take extra steps to make them be visible.

**Comment by medmunds** [comment]
It *seems like* conversations are only hidden this way in the "Conversation" tab—not in the "Files changed" tab.[^1] So a workaround is to always check a PR's "Files changed" comments before moving forward with a merge.

[^1]: Or it's random, and I just got lucky that the last several PR's I've looked at with hidden conversations didn't hide anything in "Files changed."

**Reply by ChrisMacGregorONI** [reply]
True, that is somewhat better than nothing, but it's very awkward to have to step through all of the conversations in a tiny popup (about 400x365, less than 1.8% of the area of a 4K display, no matter how I resize the window ☹️) that shows only a filename (no line numbers) and an overly-truncated text snippet for each conversation.  And of course as soon as I click on one of them, I've lost my place in the list and have to scroll to find it.

As others have suggested in some of the several other discussions of this problem (see below), having buttons to step to the "Previous [unresolved] conversation" and "Next [unresolved] conversation" would be another way to ease the pain here.

GH Community discussion threads on this that I've happened to see links to (so there are probably more):
- https://github.com/orgs/community/discussions/7638
- https://github.com/orgs/community/discussions/11842
- https://github.com/orgs/community/discussions/16111
- https://github.com/orgs/community/discussions/75820
- https://github.com/orgs/community/discussions/112603
- https://github.com/orgs/community/discussions/130618 (this one)

**Comment by DanielGibson** [comment]
Yes, this is super annoying.

What makes this even extra annoying:
Apparently those little notices about added commits are counted as posts, so if you rebase and forcepush your branch, it can happen that only these commits and maybe one recent comment show up at the end of the list (after the hidden items). So you'll see a lot of old comments at the top and basically no new comments at the bottom.
To add insult to injury, there is only one "Load more…" button and it *only loads more **old** comments*, so you have to click it several times until *all* comments are loaded, just to see more than one or two recent ones.

Ideally you **either** completely remove this feature *(maybe fold those commit notifications into "$user added $N commits" instead, if it's more than two new commits or whatever)* **or** add two additional buttons so you have three buttons in the "hidden items" section:

- `[Load more old replies…]`
- `[Load all hidden replies]`
- `[Load more new replies…]`

where the last one adds N more of the newest replies.

I'm really surprised that this was deployed to endusers in the current state **and** also hasn't been fixed/improved in several years (despite several threads here complaining about it).
Does anyone at Github actually use Github?

**Comment by stanislav-tkach** [comment]
I also find this feature disruptive. It wouldn't be that bad if at least only resolved conversations would be hidden. I'm often facing situations when Github shows some resolved and outdated comments and hides the important ones that still need to be fixed or discussed.

**Comment by LeleDallas** [comment]
I created a Chrome extension that streamlines PR reviews by hiding unnecessary timeline entries, helping you focus on what truly matters. It also reduces scrolling, making it easier to find the "Load more" button when needed.
You can find it [here](https://github.com/LeleDallas/Conversation-Cleaner)

**Reply by stanislav-tkach** [reply]
Can you please create such extension for Firefox too?..

**Reply by LeleDallas** [reply]
> Can you please create such extension for Firefox too?..

I can try! Maybe open an issue in the repo as reminder 😉

**Reply by stanislav-tkach** [reply]
Thanks in advance and here is the issue: https://github.com/LeleDallas/Conversation-Cleaner/issues/2

**Comment by paul-fastly** [comment]
Adding another vote for "please stop hiding random conversations" in PRs. I'm missing important conversations that are still in progress because GitHub decided to hide them for some reason.

**Comment by luisschwab** [comment]
Agree, this is very annoying. I can't directly CTRL+F something, and have to instead scroll until I find the button, click it and then CTRL+F something. Just add a toggle on settings to disable this.

**Reply by LeleDallas** [reply]
I've built and updated a chrome extension that include an ["Expand All"](https://github.com/LeleDallas/Conversation-Cleaner/pull/6) button in the popup. It will automatically click all "Load more…" buttons so you can quickly reveal all content and use Ctrl+F without scrolling manually

**Comment by Pierre-Bartet** [comment]
Classic enshitification, hasn't been solved for years despite tens of issues and hundreds of upvotes.

**Comment by SkohTV** [comment]
Bumping this
Working with issues of 1k+ comments is awfull atm

**Comment by azcguy** [comment]
Why are you complaining? Github knows better what you need and need not see. They are a benevolent patron that helps us little people, guides us, spares us the burden of reading posts and scrolling those poor mouse wheels. Leave them alone, they are doing what they can to help you.
